lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java

```diff
@@ -989,6 +995,70 @@ public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter reco
         return succinctBytes(outputFile.length());
     }
 
+    public static void writeOrcHiveSingleStringColumn(
+            String field,
+            File outputFile,
+            OrcTester.Format format,
+            CompressionKind compression,
+            Iterator<Function<Integer, Object>> values)
+            throws Exception
+    {
+        StandardStructObjectInspector objectInspector = getStandardStructObjectInspector(ImmutableList.of(field),
+                ImmutableList.of(javaStringObjectInspector));
+
+        writeOrcHive(
+                outputFile,
+                format,
+                compression,
+                objectInspector,
+                ImmutableList.of(VARCHAR),
+                values);
+    }
+
+    public static void writeOrcHive(
+            File outputFile,
+            OrcTester.Format format,
+            CompressionKind compression,
+            StandardStructObjectInspector objectInspector,
+            List<Type> types,
+            Iterator<Function<Integer, Object>> values)
+            throws Exception
+    {
+        writeOrcHive(
+                createOrcRecordWriter(outputFile, format, compression, objectInspector),
+                objectInspector,
+                types,
+                values
+        );
+    }
+
+    public static void writeOrcHive(
+            RecordWriter recordWriter,
+            StandardStructObjectInspector objectInspector,
+            List<Type> types,
+            Iterator<Function<Integer, Object>> values)
+            throws Exception
+    {
+        Object row = objectInspector.create();
+
+        List<StructField> fields = ImmutableList.copyOf(objectInspector.getAllStructFieldRefs());
+        Serializer serializer = new OrcSerde();
+
+        int fieldCount = fields.size();
+        while (values.hasNext()) {
+            Function<Integer, Object> itemFactory = values.next();
+            for (int fieldIndex = 0; fieldIndex < fieldCount; fieldIndex++) {
+                Object value = itemFactory.apply(fieldIndex);
+                value = preprocessWriteValueHive(types.get(fieldIndex), value);
```

Comment:
we should validate that `types` has the same amount of elements as `fieldCount`

Reply:
done